### PR TITLE
feat: add types for zpl-image package

### PR DESCRIPTION
**Pull request by jlismore**
If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/zpl-image/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Type definitions for zpl-image 0.2
```
Comment (jakebailey) [inline types/zpl-image/index.d.ts:1]: This package must have been generated via an old `dts-gen`. You'll have to match the current package format (I would recommend reading the README and checking out other packages.)

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/zpl-image/zpl-image-tests.ts
1. types/zpl-image/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 59f53f2cfde2ec213866d1f57e1bbf19ef4db6b6
</p>

File: types/zpl-image/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/// <reference types="node" />
+
+export interface Options {
+    /** @default 50 */
+    black?: number;
```
Comment (jakebailey) [inline types/zpl-image/index.d.ts:5]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

Comment (jlismore) [reply types/zpl-image/index.d.ts]: Done. Out of interest why not enable `exactOptionalPropertyTypes` in tsconfig?

Comment (jlismore) [reply types/zpl-image/index.d.ts]: Done. Out of interest why not enable `exactOptionalPropertyTypes` in tsconfig?

Comment (jakebailey) [reply types/zpl-image/index.d.ts]: Not a bad question; I don't know the answer. It may catch some stuff, but they'd have to be errors in the tests.

Comment (peterblazejewicz) [reply types/zpl-image/index.d.ts]: > Done. Out of interest why not enable `exactOptionalPropertyTypes` in tsconfig?

we are coding to allow to use that property when consuming DT modules, but modules do not have those.

**Comment by peterblazejewicz** [comment]
@jakebailey this module (native JS) looks to be old UMD module:
https://github.com/metafloor/zpl-image/blob/master/zpl-image.js#L2-L14
In order to support those, PR owner would have to update the `tscofnig` and opt-out from `node16` (back to `commonjs`) and add global export.
What we want to promote, as 'best option' for future? We could stick to `node16`, and only fallback when people start submitting requests to support old format. wdyt?

**Comment by jakebailey** [comment]
Why is the module setting a problem? The tests pass; am I missing something? `export as namespace` should work in `node16` too.

**Comment by peterblazejewicz** [comment]
adding global export throws:
```console
'zplImage' refers to a UMD global, but the current file is a module. Consider adding an import instead
```
falling back to `commonjs` make the test pass (I could be on outdated pnpm deps?, I'll double check)

**Comment by jlismore** [comment]
Sorry about the wait between changes. I can't find definitive guidance on what types for a UMD package should look like and see multiple different patterns used within the DT repo for other UMD packages. Is the required change here as simple as changing `"module": "commonjs"` in tsconfig? Or should the structure of the typing file change as well?

Any suggestions @jakebailey @peterblazejewicz?

**Comment by jakebailey** [comment]
I am definitely not experienced enough to know what to do about UMD modules. We could flip back to module=commonjs, but it sure feels gross. But I think I'd need to see a change that actually added something relevant to UMD-ness.

**Comment by peterblazejewicz** [comment]
@jlismore sorry for little side topic. Keep it to 'node16'. thanks!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Pull request merged**